### ISSUE TITLE
Complicated but seemingly real address failing

### ISSUE THREAD

**Issue by mikejcorey** [datamade/usaddress#178]
ORIGINAL STRING:  3315 S General McMullen, Bldg 171, Ste 7002 San Antonio, TX 78236 US
PARSED TOKENS:    [(u'3315', 'AddressNumber'), (u'S', 'StreetNamePreDirectional'), (u'General', 'StreetName'), (u'McMullen,', 'StreetNamePostType'), (u'Bldg', 'SubaddressType'), (u'171,', 'SubaddressIdentifier'), (u'Ste', 'SubaddressType'), (u'7002', 'SubaddressIdentifier'), (u'San', 'PlaceName'), (u'Antonio,', 'PlaceName'), (u'TX', 'StateName'), (u'78236', 'ZipCode'), (u'US', 'CountryName')]
UNCERTAIN LABEL:  SubaddressType

**Comment by jeancochrane** [comment]
Thanks for filing this @mikejcorey! Seems like "Ste 7002" should be OccupancyType/OccupancyIdentifier in this case. Does that match what you were expecting?

If you have more patterns of this type (building + suite #, or just multiple subaddress types in general) we'd love to add them in as training data.

**Comment by mikejcorey** [comment]
Yes, that looks right to me. Here's all the other U.S., seemingly non-malformed, non-taggable strings from that run:

3315 S General McMullen, Bldg 171, Ste 7002 San Antonio, TX 78236 US
601 PENNSYLVANIA AVE NW STE 900 S BLDG WASHINGTON, DC 200042625 USA
11603 N Houston Rosslyn Building 2 Houston, TX 77086 US
2817 South 294th E. Ave Broken Arrow, OK 74014 US <-- This should be "East Avenue"

There were not many -- this was a run of like 900+ addresses. I'm very impressed. Thanks for being so on top of this!

**Comment by jeancochrane** [comment]
Hey @mikejcorey,

Thanks for your patience! I finally got around to this this morning.

Looks like this wasn't quite enough training data to reliably solve the pattern. Part of the problem, I think, is that each of these patterns is slightly different – for the most part they all involve buildings and suites, but in different combinations. (Usually 4-5 examples of a specific pattern is enough.) 

Normally we save one address to add to our testing data, but in this case I just added them all into the training data in [9d0f19c](https://github.com/datamade/usaddress/commit/9d0f19cb96eabcdd156b989b903703fe84dab1a6). So for now, we won't be able to verify a performance improvement – but if you run into more patterns like this, go ahead and file a new issue and we can test them!
